### ISSUE TITLE
Cannot compile

### ISSUE THREAD

**Issue by eddieaili** [twitter/bookkeeper#1]
mvn compile failing due to dependency issue, on Ubuntu 14.04 maven 3.

**Comment by sijie** [comment]
@eddieaili could you post your error messages?

**Comment by eddieaili** [comment]
bookkeeper-shims/zookeeper-shim-3.x/pom.xml does not exclude jms. Even if I fixed jms, similar dependency issue remains for 
`[ERROR] Failed to execute goal on project twitter-finagle-provider: Could not resolve dependencies for project org.apache.bookkeeper.stats:twitter-finagle-provider:jar:4.3.1-TWTTR-SNAPSHOT: Failure to find com.twitter:finagle-core_2.10:jar:6.20.1 in http://maven.twttr.com was cached in the local repository, resolution will not be reattempted until the update interval of twitter has elapsed or updates are forced -> [Help 1]
`

```
[INFO] Reactor Summary:
[INFO] 
[INFO] bookkeeper ........................................ SUCCESS [1.439s]
[INFO] compability dependencies .......................... SUCCESS [0.056s]
[INFO] bookkeeper-server-compat400 ....................... SUCCESS [8.496s]
[INFO] bookkeeper-server-compat410 ....................... SUCCESS [4.073s]
[INFO] BookKeeper Shims .................................. SUCCESS [0.010s]
[INFO] ZooKeeper Shim .................................... SUCCESS [1.149s]
[INFO] ZooKeeper 3.3 Shim Implementation ................. FAILURE [0.073s]
[INFO] ZooKeeper 3.4 Shim Implementation ................. FAILURE [0.026s]
[INFO] Stats API for bookkeeper .......................... SUCCESS [0.444s]
[INFO] bookkeeper-server ................................. FAILURE [3.117s]
[INFO] bookkeeper-benchmark .............................. FAILURE [4.984s]
[INFO] Stats provider for Finagle stats .................. FAILURE [0.268s]
[INFO] Stats provider for twitter-stats package .......... SUCCESS [6.808s]
[INFO] Stats provider for twitter-ostrich package ........ SUCCESS [6.903s]
[INFO] bookkeeper-stats-providers ........................ SUCCESS [0.034s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 38.763s
[INFO] Finished at: Thu Apr 07 05:44:47 UTC 2016
[INFO] Final Memory: 28M/69M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project zookeeper-shim-3.3: Could not resolve dependencies for project org.apache.bookkeeper.shims:zookeeper-shim-3.3:jar:4.3.1-TWTTR-SNAPSHOT: The following artifacts could not be resolved: javax.jms:jms:jar:1.1, com.sun.jdmk:jmxtools:jar:1.2.1, com.sun.jmx:jmxri:jar:1.2.1: Could not transfer artifact javax.jms:jms:jar:1.1 from/to java.net (https://maven-repository.dev.java.net/nonav/repository): No connector available to access repository java.net (https://maven-repository.dev.java.net/nonav/repository) of type legacy using the available factories WagonRepositoryConnectorFactory -> [Help 1]
[ERROR] Failed to execute goal on project zookeeper-shim-3.4: Could not resolve dependencies for project org.apache.bookkeeper.shims:zookeeper-shim-3.4:jar:4.3.1-TWTTR-SNAPSHOT: The following artifacts could not be resolved: javax.jms:jms:jar:1.1, com.sun.jdmk:jmxtools:jar:1.2.1, com.sun.jmx:jmxri:jar:1.2.1: Failure to transfer javax.jms:jms:jar:1.1 from https://maven-repository.dev.java.net/nonav/repository was cached in the local repository, resolution will not be reattempted until the update interval of java.net has elapsed or updates are forced. Original error: Could not transfer artifact javax.jms:jms:jar:1.1 from/to java.net (https://maven-repository.dev.java.net/nonav/repository): No connector available to access repository java.net (https://maven-repository.dev.java.net/nonav/repository) of type legacy using the available factories WagonRepositoryConnectorFactory -> [Help 1]
[ERROR] Failed to execute goal on project bookkeeper-server: Could not resolve dependencies for project org.apache.bookkeeper:bookkeeper-server:jar:4.3.1-TWTTR-SNAPSHOT: Could not find artifact org.apache.bookkeeper.shims:zookeeper-shim-3.4:jar:4.3.1-TWTTR-SNAPSHOT in apache.snapshots (http://repository.apache.org/snapshots) -> [Help 1]
[ERROR] Failed to execute goal on project bookkeeper-benchmark: Could not resolve dependencies for project org.apache.bookkeeper:bookkeeper-benchmark:jar:4.3.1-TWTTR-SNAPSHOT: The following artifacts could not be resolved: com.twitter.common:stats:jar:0.0.51, org.apache.bookkeeper:bookkeeper-server:jar:4.3.1-TWTTR-SNAPSHOT, org.apache.bookkeeper.shims:zookeeper-shim-3.4:jar:4.3.1-TWTTR-SNAPSHOT, org.apache.bookkeeper:bookkeeper-server:jar:tests:4.3.1-TWTTR-SNAPSHOT: Could not find artifact com.twitter.common:stats:jar:0.0.51 in central (http://repo.maven.apache.org/maven2) -> [Help 1]
[ERROR] Failed to execute goal on project twitter-finagle-provider: Could not resolve dependencies for project org.apache.bookkeeper.stats:twitter-finagle-provider:jar:4.3.1-TWTTR-SNAPSHOT: Could not find artifact com.twitter:finagle-core_2.10:jar:6.20.1-TWOTEN-1407871523 in twitter (http://maven.twttr.com) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :zookeeper-shim-3.3
[INFO] Build failures were ignored.
```

**Comment by sijie** [comment]
@eddieaili I found the problem. we need to fix the finagle_core version.

**Comment by sijie** [comment]
@eddieaili I would try to upload the fix on coming Monday

**Comment by sijie** [comment]
@eddieaili we fixed the dependency issue in 8c2d55b9a51a13b5d48ce91eeb1278a2e74839ef. could you retry it?
